src/OAuth/OAuth1/Service/Etsy.php

```diff
@@ -27,12 +52,19 @@ public function __construct(
         }
     }
 
+    public function setScopes($requested_scopes = array())
+    {
+        if ($requested_scopes) {
```

Comment:
This logic doesn't make sense. The default argument is no scopes, but if there are no scopes nothing will actually change.

Is it valid to have no scopes? If so, how would you set that? It should look something like this:

```php
if ($requested_scopes) {
    $this->scopes = '?scope=' . implode('%20', $requested_scopes);
} else {
    $this->scopes = '';
}
```